File: leetcode/easy/0104MaxDepth/golang/depth.go
package golang

// TreeNode Definiton
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxDepthRecursive(root *TreeNode) int {
	if root == nil {
		return 0
	}
	leftDepth := maxDepthRecursive(root.Left)
	rightDepth := maxDepthRecursive(root.Right)
	return findMax(leftDepth, rightDepth) + 1
}

func findMax(numberOne int, numberTwo int) int {
	if numberOne > numberTwo {
		return numberOne
	}
	return numberTwo
}

File: leetcode/easy/0104MaxDepth/depth.kt
data class TreeNode(var `val`: Int) {
    var left: TreeNode? = null
    var right: TreeNode? = null
}

class RecursiveSolution {
    fun maxDepth(root: TreeNode?): Int {
        if(root == null){
            return 0
        }
        var leftDepth = maxDepth(root.left)
        var rightDepth = maxDepth(root.right)

        if(leftDepth > rightDepth){
            return leftDepth + 1
        }else{
            return rightDepth + 1
        }
    }
}